docs/presentation/slides.md

```diff
@@ -0,0 +1,507 @@
+  smoothing to move the power slowly towards a target power value to avoid jerky movement.
+
+---
+
+# Debugging electronics
+
+![PWM signals on an oscilloscope](scope.jpg)
+
+???
+
+- Blue and yellow are control signals for two halves of one channel running at 50% duty cycle.
+- Note the slight offset in timing; it's better to have the motor be undriven for a fraction of a
+  second than have a short circuit. The PWM hardware is configured to do this automatically.
+- Pink is another channel. Scale is different because the scope only has two analogue inputs, this
+  is a digital input.
+
+---
+
+# Where to?
+
+What we actually want to do is pull a kite string, so we care about position more than speed.
+
+```rust
+if let Some(target_position) = target_position {
+  let difference = target_position - position;
+  speed = clamp(difference * spring_constant, &speed_limits.into());
+} else {
+  speed = 0;
+}
+
+hoverboard.set_motor_power(speed);
+```
+
+???
+
+- Yet another layer sets the target power based on the absolute position and a target position. The
+  power is set to the difference times a configured spring constant, capped at a configured power
+  limit.
+- Experimentally, trying to turn the wheel does indeed feel like a spring!
+- We also set the LEDs based on the position difference, to give a visual indication of what's happening.
+
+---
+
+# Talking to a computer (and the other half)
+
+- Each side has two USARTs we can use.
+- For ease of use, we connect the 'primary' side to the computer, and the secondary side to the
+  primary side. The primary side relays commands to the secondary side.
+- Command crate shared between device and controller sides:
+
+```rust
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Command {
+  SetSideLed(bool),
+  SetOrangeLed(bool),
+  SetRedLed(bool),
+  SetGreenLed(bool),
+  AddBuzzerNote(Note),
+  ReportBattery,
+  ReportCharger,
+  SetMaxSpeed(SpeedLimits),
+  SetSpringConstant(u16),
+  SetTarget(i64),
+  RemoveTarget,
+  Recenter,
+  IncrementTarget,
+  DecrementTarget,
+  PowerOff,
+  TestMotor,
+}
+```
+
+???
+
+- This can be connected either to a laptop via a USB serial adapter, or directly to a Raspberry Pi.
+- Parsing is a big `match` expression, matching on a slice.
+- Parsing and serialisation code has plenty of tests, which can be run on the host.
+- The eventual plan is to have a webcam plugged into a Raspberry Pi or laptop, with some computer
+  vision code seeing where the kite is in the sky and which way it is pointing, and use that to run
+  a control system.
+- For now, we have some code to use a game controller plugged into the laptop or Raspberry Pi to
+  control it manually.
+
+---
+
+![Set up at Hampstead Heath](field.jpg)
+
+Ready to go
+
+---
+
+![Flying at Hampstead Heath](flying.jpg)
+
+In the air
+
+---
+
+![Telemetry in Grafana](telemetry.png)
+
+???
+
+- Telemetry is sent over MQTT, from where it can be monitored and logged to InfluxDB for later
```

Comment:
```suggestion
- Telemetry is sent from the desktop app over MQTT to a local stats server, from where it can be monitored and logged to InfluxDB for later
```

Reply:
Done.